### DISCUSSION TITLE
Ability to add in a custom cover image

### DISCUSSION THREAD

**Discussion by Onkeyuk** [kudrykv/latex-yearly-planner#81] [Ideas]
I would love a way to add a page at the start where we can add a custom cover image or pull nice repeating patterns together to pick from with a title area that gets the name. Eg Daily Planner 2023

**Comment by philipsd6** [comment]
FWIW, I've been doing this to "underlay" a custom image under the year numbers on the cover with `qpdf`:

```sh
qpdf rm2.mos.default.dailycal.2023.pdf --underlay cover.pdf --to=1 --replace-input
```
![photo_2023-05-02_09-58-26](https://user-images.githubusercontent.com/908201/235690906-e502535f-dda9-4931-af2d-6d237dd6427a.jpg)
I made that graphic by customizing a [Hero Pattern](https://heropatterns.com/) in [GIMP](https://www.gimp.org/). It's important to use the `--underlay` feature because that maintains the existing links between pages and simply removing the cover and replacing it with a custom page destroys the links.

**Reply by Onkeyuk** [reply]
That’s awesome. Wish I understood how to generate custom calendars though, I’m just using pre built ones

**Reply by kudrykv** [reply]
@Onkeyuk generating a template _should_ be as simple as running this command (assuming you have latest Go and LaTeX installed):
https://github.com/kudrykv/latex-yearly-planner/discussions/59#discussioncomment-3780843

Then, all configurable bits that available now are mentioned in the command, inside specified files

**Reply by philipsd6** [reply]
> That’s awesome. Wish I understood how to generate custom calendars though, I’m just using pre built ones 

You can use my method with the prebuilt ones too!

**Comment by kudrykv** [comment]
That sounds interesting.

I'm massively reworking the code, so b/c of that can't promise when it'll be ready and in which way, though.

The underlay option that @philipsd6 suggests is a cool way to overcome this

**Comment by Onkeyuk** [comment]
Thanks both. I’m on my mobile at the Moment so I’ll have a google later. I currently don’t have go installed or anything setup, just a copy of the pdf so I’ll work it out this evening. 
I’m assuming qpdf is a pdf creation tool which I haven’t heard of. I’ll have play
